File: sorting/selection-sort/selection-sort.ts
function selectionSort(numbers: Array<number>): Array<number> {
    for (let i = 0; i < numbers.length; i++) {
        let minIndex = i;
        
        for (let j = i; j < numbers.length; j++) {
            if (numbers[j] < numbers[minIndex]) {
                minIndex = j;
            }
        }

        const temp = numbers[i];
        numbers[i] = numbers[minIndex];
        numbers[minIndex] = temp;
    }

    return numbers;
}

// Test
console.log(selectionSort([5,3,7,2,9,1]));

File: sorting/selection-sort/selection_sort.py
def selection_sort(numbers):
    index = 0

    while index < len(numbers):
        minIndex = index

        for idx in range(index, len(numbers)):
            if (numbers[idx] < numbers[minIndex]):
                minIndex = idx

        temp = numbers[index]
        numbers[index] = numbers[minIndex]
        numbers[minIndex] = temp
        index += 1
    
    return numbers

# Test
print(selection_sort([5,3,7,2,9,1]))
